source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -317,17 +314,27 @@ RedisClusterFactory::createClusterWithConfig(
     Envoy::Stats::ScopePtr&& stats_scope) {
   if (!cluster.has_cluster_type() ||
       cluster.cluster_type().name() != Extensions::Clusters::ClusterTypes::get().Redis) {
-    throw EnvoyException("Redis cluster can only created with redis cluster type");
+    throw EnvoyException("Redis cluster can only created with redis cluster type.");
+  }
+  // TODO(hyang): This is needed to migrate existing cluster, disallow using other lb_policy
```

Comment:
This feature is not complete or fully working yet. I don't think we need to deal with back compat? I would just switch it?

Reply:
We're currently using it with Random, so if we don't do back compat then we'll have to make sure both the config changes and the envoy binary are roll out at the same time.  It's better to do this and remove it after both the config changes and binary are roll out.